File: src/src_core/fmg/core/img/RotateTransformer.cs
using System;
using fmg.common.ui;

namespace fmg.core.img {

   /// <summary> Transforming of rotate angle </summary>
   public class RotateTransformer : IModelTransformer {

      public void Execute(IAnimatedModel model) {
         if (!(model is AnimatedImageModel am))
            throw new Exception("Illegal usage transformer");

         double rotateAngleDelta = 360.0 / am.TotalFrames; // 360° / TotalFrames
         if (!am.AnimeDirection)
            rotateAngleDelta = -rotateAngleDelta;
         double rotateAngle = am.CurrentFrame * rotateAngleDelta;
         am.RotateAngle = rotateAngle;
      }

   }

}

File: src/src_core/fmg/core/img/RotateTransformer.js
/** Transforming of rotate angle */
class RotateTransformer extends IModelTransformer {

    static get className() { return 'RotateTransformer'; }
    get className() { return RotateTransformer.className; }

    execute(/*IImageModel*/ model) {
        var am = model;

        var rotateAngleDelta = 360.0 / am.totalFrames; // 360° / TotalFrames
        if (!am.animeDirection)
            rotateAngleDelta = -rotateAngleDelta;
        var rotateAngle = am.currentFrame * rotateAngleDelta;
        am.rotateAngle = rotateAngle;
    }

}
